pkg/authorization/api/v1/conversion.go

```diff
@@ -260,171 +264,180 @@ func convert_api_ClusterRoleBinding_To_v1_ClusterRoleBinding(in *newer.ClusterRo
 	return nil
 }
 
-func convert_v1_ClusterPolicyBinding_To_api_ClusterPolicyBinding(in *ClusterPolicyBinding, out *newer.ClusterPolicyBinding, s conversion.Scope) error {
+func Convert_v1_ClusterPolicyBinding_To_api_ClusterPolicyBinding(in *ClusterPolicyBinding, out *newer.ClusterPolicyBinding, s conversion.Scope) error {
 	out.LastModified = in.LastModified
 	out.RoleBindings = make(map[string]*newer.ClusterRoleBinding)
 	return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
 }
 
-func convert_api_ClusterPolicyBinding_To_v1_ClusterPolicyBinding(in *newer.ClusterPolicyBinding, out *ClusterPolicyBinding, s conversion.Scope) error {
+func Convert_api_ClusterPolicyBinding_To_v1_ClusterPolicyBinding(in *newer.ClusterPolicyBinding, out *ClusterPolicyBinding, s conversion.Scope) error {
 	out.LastModified = in.LastModified
 	out.RoleBindings = make([]NamedClusterRoleBinding, 0, 0)
 	return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
 }
 
+func Convert_v1_NamedRoleArray_to_api_RoleArray(in *[]NamedRole, out *map[string]*newer.Role, s conversion.Scope) error {
+	for _, curr := range *in {
+		newRole := &newer.Role{}
+		if err := s.Convert(&curr.Role, newRole, 0); err != nil {
+			return err
+		}
+		(*out)[curr.Name] = newRole
+	}
+
+	return nil
+}
+func Convert_api_NamedRoleArray_to_v1_RoleArray(in *map[string]*newer.Role, out *[]NamedRole, s conversion.Scope) error {
+	allKeys := make([]string, 0, len(*in))
+	for key := range *in {
+		allKeys = append(allKeys, key)
+	}
+	sort.Strings(allKeys)
+
+	for _, key := range allKeys {
+		newRole := (*in)[key]
+		oldRole := &Role{}
+		if err := s.Convert(newRole, oldRole, 0); err != nil {
+			return err
+		}
+
+		namedRole := NamedRole{key, *oldRole}
+		*out = append(*out, namedRole)
+	}
+
+	return nil
+}
+
+func Convert_v1_NamedRoleBindingArray_to_api_RoleBindingArray(in *[]NamedRoleBinding, out *map[string]*newer.RoleBinding, s conversion.Scope) error {
+	for _, curr := range *in {
+		newRoleBinding := &newer.RoleBinding{}
+		if err := s.Convert(&curr.RoleBinding, newRoleBinding, 0); err != nil {
```

Comment:
Does `Convert` also do defaulting?  I *think* it does, but it seems reasonable to ask.